packages/sdk/src/util/planarGridLayout.ts

```diff
@@ -0,0 +1,190 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Actor, AnimationEaseCurves, BoxAlignment, InvertBoxAlignment, Vector3 } from '..';
+
+/** Options for [[GridLayout.addCell]]. */
+export interface AddCellOptions {
+	/** The actor to be placed in the grid cell. Must be parented to the grid root. */
+	contents: Actor;
+	/** The row index, with 0 at the top. */
+	row: number;
+	/** The column index, with 0 on the left. */
+	column: number;
+	/** The width of this cell for layout purposes. Should include any desired padding. */
+	width: number;
+	/** The height of this cell for layout purposes. Should include any desired padding. */
+	height: number;
+	/** Where the actor should be placed within the cell box. Defaults to [[GridLayout.defaultCellAlignment]]. */
+	alignment?: BoxAlignment;
+}
+
+const sumFn = (sum: number, x: number) => sum + x;
+const maxFn = (max: number, x: number) => Math.max(max, x);
+
+/**
+ * Lay out actors in a grid along the root actor's local XY plane. Assign actors to the grid with [[addCell]],
+ * and apply updates with [[applyLayout]].
+ */
+export class PlanarGridLayout {
+	private contents: AddCellOptions[] = [];
+
+	/**
+	 * Initialize a new grid layout.
+	 * @param anchor The grid's anchor actor, the point to which the grid is aligned.
+	 * @param gridAlignment How the grid should be aligned to its anchor, where [[BoxAlignment.TopLeft]] will place
+	 * the grid above and to the left of the anchor, and the lower right corner will touch the anchor.
+	 * @param defaultCellAlignment How cells should be aligned by default.
+	 */
+	public constructor(
+		private anchor: Actor,
+		public gridAlignment = BoxAlignment.MiddleCenter,
+		public defaultCellAlignment = BoxAlignment.MiddleCenter
+	) { }
+
+	/** The number of columns in this grid. */
+	public columnCount() {
```

Comment:
>columnCount [](start = 8, length = 11)

I suggest making this and the rest getters. Or, prefix the method name with `get` --> `getColumnCount`